### PR TITLE
Port splitCheck and SplitCheckObserver test.

### PR DESCRIPTION
**Pull request by xiaobogaga**
@coocood PTAL, thanks.

### CODE REVIEW

File: tikv/raftstore/coprocessor_test.go [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+		Id: 1,
+		Peers: []*metapb.Peer{
+			new(metapb.Peer),
+		},
+		RegionEpoch: &metapb.RegionEpoch{
+			ConfVer: 5,
+			Version: 2,
+		},
+		StartKey: codec.EncodeInt(tablecodec.TablePrefix(), int64(0)),
+		EndKey:   codec.EncodeInt(tablecodec.TablePrefix(), int64(0)),
```
Comment (coocood) [inline tikv/raftstore/coprocessor_test.go]: The end key must be greater than start key unless both of them are empty.
Here the initial start key should be `[]byte{'t'}`, the initial end key should be `[]byte{'u'}`.

File: tikv/raftstore/coprocessor_test.go [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+	}
+}
+
+func TestKeysSplitCheck(t *testing.T) {
+	engines := newTestEngines(t)
+	defer cleanUpTestEngineData(engines)
+
+	config := newDefaultSplitCheckConfig()
+	config.regionMaxKeys = 100
+	config.regionSplitKeys = 90
```
Comment (coocood) [inline tikv/raftstore/coprocessor_test.go]: It is `80` in tikv.

File: tikv/raftstore/coprocessor_test.go [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+		assert.Equal(t, splitRegion.RegionEpoch, expectRegion.RegionEpoch)
+		assert.Equal(t, splitRegion.SplitKeys, expectSplitKeys)
+	} else {
+		panic(fmt.Sprintf("expect split check result. but got %v", msg))
+	}
+}
+
+func putData(db *DBBundle, startIdx, endIdx uint64) {
+	val := []byte("some value")
+	arr := [1024]byte{}
```
Comment (coocood) [inline tikv/raftstore/coprocessor_test.go]: Why not `make([]byte, 1024)`?

File: tikv/raftstore/coprocessor_test.go [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+	for startIdx < endIdx {
+		batchIdx := startIdx + 20
+		if batchIdx > endIdx {
+			batchIdx = endIdx
+		}
+		writeBatch := new(WriteBatch)
+		for i := startIdx; i < batchIdx; i++ {
+			key := DataKey(codec.EncodeBytes(nil, []byte(fmt.Sprintf("%04d", i))))
+			writeBatch.Set(key, val)
+			writeBatch.Set(key, arr[:])
```
Comment (coocood) [inline tikv/raftstore/coprocessor_test.go]: We don't need to set the same key twice.

File: tikv/raftstore/coprocessor_test.go [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+		}
+		// Flush to generate SST files, so that properties can be utilized.
```
Comment (coocood) [inline tikv/raftstore/coprocessor_test.go]: We don't have properties in SST, so we can just write a single batch.

File: tikv/raftstore/coprocessor_test.go [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+			autoSplit: autoSplit,
+			policy:    policy,
+		},
+	})
+}
+
+func mustSplitAt(t *testing.T, rx <-chan Msg, expectRegion *metapb.Region, expectSplitKeys [][]byte) {
+	msg := <-rx
+	if msg.Type == MsgTypeRegionApproximateSize || msg.Type == MsgTypeRegionApproximateKeys {
+		assert.Equal(t, msg.RegionID, expectRegion.Id)
```
Comment (coocood) [inline tikv/raftstore/coprocessor_test.go]: Maybe the test passed because all messsages run into this branch.

**Comment by xiaobogaga** [comment]
@coocood PTAL again, thanks.

**Comment by xiaobogaga** [comment]
some test are removed, because they relies on the approximation related  strategies such as `getRegionApproximateMiddle`, while these strategies are not implemented right now.

**Comment by coocood** [comment]
LGTM

**Pull request merged**